test/common/http/http1/conn_pool_test.cc

```diff
@@ -540,5 +541,41 @@ TEST_F(Http1ConnPoolImplTest, DrainCallback) {
   dispatcher_.clearDeferredDeleteList();
 }
 
+TEST_F(Http1ConnPoolImplTest, RemoteClose) {
+  InSequence s;
+
+  NiceMock<Http::MockStreamDecoder> outer_decoder;
+  ConnPoolCallbacks callbacks;
+  conn_pool_.expectClientCreate();
+  Http::ConnectionPool::Cancellable* handle = conn_pool_.newStream(outer_decoder, callbacks);
+  EXPECT_NE(nullptr, handle);
+
+  NiceMock<Http::MockStreamEncoder> request_encoder;
+  Http::StreamDecoder* inner_decoder;
+  EXPECT_CALL(*conn_pool_.test_clients_[0].codec_, newStream(_))
+      .WillOnce(DoAll(SaveArgAddress(&inner_decoder), ReturnRef(request_encoder)));
+  EXPECT_CALL(callbacks.pool_ready_, ready());
+
+  EXPECT_CALL(*conn_pool_.test_clients_[0].connect_timer_, disableTimer());
+  conn_pool_.test_clients_[0].connection_->raiseEvents(Network::ConnectionEvent::Connected);
+
+  callbacks.outer_encoder_->encodeHeaders(TestHeaderMapImpl{}, false);
+  Buffer::OwnedImpl dummy_data("12345");
+  callbacks.outer_encoder_->encodeData(dummy_data, false);
+
+  Buffer::OwnedImpl empty_data;
+  EXPECT_CALL(*conn_pool_.test_clients_[0].codec_, dispatch(BufferEqual(&empty_data)))
+      .WillOnce(Invoke([&](Buffer::Instance& data) -> void {
```

Comment:
Here you should verify that data length is 0, which is what happens in this case.